pyproject.toml

```diff
@@ -23,7 +23,8 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
 ]
 dependencies = [
-	"jupyterlab~=3.6"
+	"jupyterlab~=3.6",
+    "jupyterlab_widgets~=3.0"
```

Comment:
Should it be `ipywidgets` ?

Reply:
Yes, that's right @jtpio.

Reply:
Does it actually needs any of those two?

It seems to me only the front-end would need this dependency. If ipywidgets is not installed by the user they cannot use widgets, and that's fine. If it is installed, they can.

Reply:
If I understood correctly, jupyterlab_widgets is a lab extension. If that's the case, we need to make sure the extension is installed, and we can only do that if we depend on its Python package.

I don't understand why the versions differ between its JS and Python packages though.
https://github.com/jupyter-widgets/ipywidgets/blob/511663a56324cea5324f49a65ebe25e2f1b04d87/python/jupyterlab_widgets/jupyterlab_widgets/_version.py#L4
https://github.com/jupyter-widgets/ipywidgets/blob/511663a56324cea5324f49a65ebe25e2f1b04d87/python/jupyterlab_widgets/package.json#L3

Reply:
> It seems to me only the front-end would need this dependency. If ipywidgets is not installed by the user they cannot use widgets, and that's fine. If it is installed, they can.

Actually, I think you are right. I can require its plugging as optional and it will work once the user installs ipywidgets.

Reply:
It's the same situation as in Voila. Voila front-end requires ipywidgets JS packages (to implement our own widget manager), but the Python package does not require ipywidgets or jupyterlab_widgets.

> I don't understand why the versions differ between its JS and Python packages though

Legacy? We're in the same situation in many widgets packages (like bqplot and ipyleaflet), because at some point we only bumped the Python package because there were no applied changes in the front-end code.

Reply:
Btw, I just tried to remove the Python dependency, but I can not. 

I require `IJupyterWidgetRegistry from '@jupyter-widgets/base'` as an optional token, and still, if the token is missing, the extension doesn't activate.

Reply:
> I require IJupyterWidgetRegistry from '@jupyter-widgets/base' as an optional token, and still, if the token is missing, the extension doesn't activate.

Even if making the token optional?

Reply:
> Even if making the token optional?

Yes

> Could this be part of a separate plugin just for widgets? This way it would not prevent the rest of the extension to activate.

I think that might work. But I would need to create multiple plugins. We are not providing the document tracker at the moment.

Reply:
It's really really really annoying that a missing optional token would broke the extension.  That completely defeats the purpose of optional tokens.